src/foam/dao/journal/PutRemoveJournalDAODecorator.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao.journal',
+  name: 'PutRemoveJournalDAODecorator',
```

Comment:
What is this for?

Reply:
This is the most basic DAO decorator for all of this. It's what deals with telling the JournalEntryDAO about a put or a remove by putting a JournalEntry in the context under "journalEntry". It doesn't care about comments, diffing, hasing, etc. It's super simple and using this alone should achieve a working journal.